lib/pbench/test/unit/server/test_datasets_list.py

```diff
@@ -596,15 +636,47 @@ def test_user_no_auth(self, monkeypatch, db_session):
         assert e.value.http_status == HTTPStatus.UNAUTHORIZED
 
     @pytest.mark.parametrize(
-        "meta,error",
+        "meta,error,message",
         [
-            ("user.foo:1", HTTPStatus.UNAUTHORIZED),
-            ("x.y:1", HTTPStatus.BAD_REQUEST),
-            ("global.x=3", HTTPStatus.BAD_REQUEST),
-            ("dataset.notright:10", HTTPStatus.BAD_REQUEST),
+            (
+                "user.foo:1",
+                HTTPStatus.UNAUTHORIZED,
+                "Metadata key user.foo cannot be used by an unauthenticated client",
+            ),
+            ("x.y:1", HTTPStatus.BAD_REQUEST, "Metadata key 'x.y' is not supported"),
+            (
+                "global.x=3",
+                HTTPStatus.BAD_REQUEST,
+                "Missing terminator for 'global.x=3'",
```

Comment:
"Missing terminator" isn't as clear as it might be -- that is, there is no indication what the terminator is nor where in `global.x=3` it should have been found.

Reply:
Ha: this one is the syntax you'd like to succeed. Yeah, I suppose "missing `:` between key and value" might be better?